test/contact_properties.js

```diff
@@ -1,164 +1,296 @@
-    it('can delete', function() {
-      return hubspot.contacts.properties
-        .upsert(testDeleteProperty)
-        .then(data => {
-          expect(data).to.be.an('object')
-          expect(data).to.have.a.property('name')
-          return hubspot.contacts.properties.delete(testDeleteProperty.name)
+      it('should create the property', function() {
+        return hubspot.contacts.properties.upsert(properties).then(data => {
+          expect(data.description).to.eq(properties.description)
         })
+      })
+    })
+
+    describe('when the record already exists', function() {
+      const description = 'Updated display name'
+      const createEndpoint = {
+        path: '/properties/v1/contacts/properties',
+        statusCode: 409,
+      }
+      const updateEndpoint = {
+        path: `/properties/v1/contacts/properties/named/${properties.name}`,
+        response: { ...properties, description },
+      }
+      fakeHubspotApi.setupServer({
+        postEndpoints: [createEndpoint],
+        putEndpoints: [updateEndpoint],
+      })
+
+      before(function() {
+        if (process.env.NOCK_OFF) {
+          return createTestProperty(properties)
+        }
+      })
+      after(function() {
+        if (process.env.NOCK_OFF) {
+          return deleteTestProperty(properties.name)
+        }
+      })
+
+      it('should update the property', function() {
+        return hubspot.contacts.properties
+          .upsert({ ...properties, description })
+          .then(data => {
+            expect(data.description).to.eq(description)
+          })
+      })
     })
   })
 
-  describe('update', function() {
-    property.label = 'MadKudo Customer Fit'
+  describe('groups', () => {
+    describe('getGroups', () => {
+      const getGroupsEndpoint = {
+        path: '/properties/v1/contacts/groups',
+        response: [],
+      }
+      fakeHubspotApi.setupServer({ getEndpoints: [getGroupsEndpoint] })
 
-    it('should update the property', function() {
-      return hubspot.contacts.properties
-        .update(property.name, property)
-        .then(data => {
-          expect(data).to.be.an('object')
-          expect(data).to.have.a.property('name')
-          expect(data.label).to.equal(property.label)
+      it('should return groups', function() {
+        return hubspot.contacts.properties.getGroups().then(data => {
+          expect(data).to.be.an('array')
         })
+      })
+    })
+
+    describe('createGroup', function() {
+      const name = 'test_group'
+      const createGroupEndpoint = {
+        path: '/properties/v1/contacts/groups',
+        response: {},
+      }
+      fakeHubspotApi.setupServer({ postEndpoints: [createGroupEndpoint] })
+
+      after(function() {
+        if (process.env.NOCK_OFF) {
+          return deleteTestGroup(name)
+        }
+      })
+
+      it('returns a 200', function() {
+        return hubspot.contacts.properties
+          .createGroup({ name, displayName: 'Test Group' })
```

Comment:
I believe a `displayName` is not required to create a Contact Property Group. There's no harm in leaving it in but you could also remove it so the test focuses on only the fields that are required.